src/custom/hooks/useUSDCPrice/index.ts

```diff
@@ -100,3 +107,105 @@ export function useUSDCValue(currencyAmount?: CurrencyAmount<Currency>) {
     }
   }, [currencyAmount, error, price])
 }
+
+/**
+ * Returns the price in USDC of the input currency from price APIs
+ * @param currencyAmount currency to compute the USDC price of
+ */
+export function useUSDCValue(currencyAmount?: CurrencyAmount<Currency>) {
+  const usdcPrice = useUSDCPrice(currencyAmount?.currency)
+
+  return useGetPriceQuote({ ...usdcPrice, currencyAmount })
+}
+
+export function useCoingeckoUsdPrice({ tokenAddress }: Pick<CoinGeckoUsdPriceParams, 'tokenAddress'>) {
+  // default to MAINNET (if disconnected e.g)
+  const { chainId = DEFAULT_NETWORK_FOR_LISTS } = useActiveWeb3React()
+  const [price, setPrice] = useState<Price<Token, Currency> | null>(null)
+  const [error, setError] = useState<Error | null>(null)
+
+  const currency = useToken(tokenAddress)
+
+  useEffect(() => {
+    getUSDPriceQuote({
+      chainId,
+      tokenAddress,
+    })
+      .then(toPriceInformation)
+      .then((response) => {
+        if (!currency || !response?.amount) return
+
+        const { amount } = response
+        // api returns base units, we want atoms and CurrencyAmount
+        const usdParsed = tryParseAmount(amount.toString(), currency)
+        // parse failure is unlikely - type safe
+        if (!usdParsed) return
+
+        // create a new Price object
+        const usdPrice = new Price({
+          quoteAmount: CurrencyAmount.fromRawAmount(
+            usdParsed.currency,
+            // we use 1 as the denominator
+            parseUnits('1', usdParsed.currency.decimals).toString()
+          ),
+          baseAmount: usdParsed,
+        })
+
+        console.debug(
+          '[useCoingeckoUsdPrice] Best Coingecko USD price amount',
+          usdPrice.toSignificant(12),
+          usdPrice.invert().toSignificant(12)
+        )
+
+        setPrice(usdPrice)
+      })
+      .catch((error) => {
+        console.error(
+          '[useUSDCPrice::useCoingeckoUsdPrice]::Error getting USD price from Coingecko for token',
+          tokenAddress,
+          error
+        )
+        return batchedUpdate(() => {
+          setError(new Error(error))
+          setPrice(null)
+        })
+      })
+  }, [chainId, currency, tokenAddress])
+
+  return { price, error }
+}
+
+type CoinGeckoUsdValueParams = Pick<CoinGeckoUsdPriceParams, 'tokenAddress'> & {
+  currencyAmount?: CurrencyAmount<Currency>
+}
+
+export function useCoingeckoUSDValue(params: CoinGeckoUsdValueParams) {
+  const { currencyAmount } = params
+  const coingeckoUsdPrice = useCoingeckoUsdPrice(params)
+
+  return useGetPriceQuote({ ...coingeckoUsdPrice, currencyAmount })
+}
+
+export function useHigherUSDValue(currencyAmount: CurrencyAmount<Currency> | undefined) {
+  const usdcValue = useUSDCValue(currencyAmount)
+  const coingeckoUsdPrice = useCoingeckoUSDValue({
+    tokenAddress: currencyAmount ? currencyId(currencyAmount.currency) : '',
+    currencyAmount,
+  })
```

Comment:
Do we have any kind of timeout logic? probably not for this PR, also less important than other things, but maybe good to track it.